### PR TITLE
EDGEDEMATIC-9 - Parsing API token information

### PR DESCRIPTION
**Pull request by khandramai**
[EDGDEMATIC-9](https://issues.folio.org/browse/EDGDEMATIC-9) - Parsing API token information

## Purpose
Spring-based Edge modules should use API keys and edge users.

### TODOS and Open Questions
 - [ ] Verification of AWS integration.

## Pre-Merge Checklist:
 Before merging this PR, please go through the following list and take appropriate actions.

 - Does this PR meet or exceed the expected quality standards?
   - [ ] Code coverage on new code is 80% or greater
   - [ ] Duplications on new code is 3% or less
   - [ ] There are no major code smells or security issues
 - Does this introduce breaking changes?
   - [ ] Were any API paths or methods changed, added or removed?
   - [ ] Were there any schema changes?
   - [ ] Did any of the interface versions change?
   - [ ] Were permissions changed, added, or removed?
   - [ ] Are there new interface dependencies?
   - [ ] There are no breaking changes in this PR.

 If there are breaking changes, please **STOP** and consider the following:

 - What other modules will these changes impact?
 - Do JIRAs exist to update the impacted modules?
   - [ ] If not, please create them
   - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
   - [ ] Do they have all they appropriate links to blocked/related issues?
 - Are the JIRAs under active development?
   - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
 - Do PRs exist for these changes?
   - [ ] If so, have they been approved?

 Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

 While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by alekGbuz — APPROVED** [review]

File: src/main/java/org/folio/ed/controller/ApiKeyService.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+  }
+
+  public String getEdgeApiKey(ServletRequest servletRequest) {
+    for (ApiKeySource source : sources) {
+      String apiKey = null;
+      if (ApiKeySource.PARAM == source ) {
+        apiKey = getFromParam(servletRequest);
+      } else if (ApiKeySource.HEADER == source) {
+        apiKey = getFromHeader(servletRequest);
```
Comment (VitaliiZahoruiko) [inline src/main/java/org/folio/ed/controller/ApiKeyService.java:43]: (minor) Can we simplify this by not using additional variable like apiKey and just return getFromParam(servletRequest) or getFromHeader(servletRequest) or null ? Also we can use map or strategy here to get rid from multiple if statements

File: src/main/java/org/folio/ed/controller/ApiKeyService.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      }
+      if (apiKey != null) {
```
Comment (VitaliiZahoruiko) [inline src/main/java/org/folio/ed/controller/ApiKeyService.java:45]: (minor) Additional check can be removed (we return apiKey or null anyway)

File: src/main/java/org/folio/ed/service/RemoteStorageService.java [hunk 3]
```diff
@@ -20,61 +21,69 @@
...
-    return remoteStorageClient.getStorageConfigurations()
+  public List<Configuration> getStagingDirectorConfigurations(String tenantId, String okapiToken) {
+
+    List<Configuration> stagingDirectorConfigurations = new ArrayList<>();
+    remoteStorageClient.getStorageConfigurations(tenantId, okapiToken)
       .getConfigurations()
-      .stream()
-      .filter(configuration -> STAGING_DIRECTOR_NAME.equals(configuration.getProviderName()))
-      .collect(Collectors.toList());
+      .forEach(c -> {
```
Comment (VitaliiZahoruiko) [inline src/main/java/org/folio/ed/service/RemoteStorageService.java:45]: (minor)Please use more meaningful names for variables

File: src/main/java/org/folio/ed/service/StagingDirectorFlowsService.java [hunk 4]
```diff
@@ -73,26 +80,35 @@ public void removeExistingFlows() {
...
     return integrationFlowContext
       .registration(IntegrationFlows
-        .from(() -> remoteStorageService.getAccessionQueueRecords(configuration.getId()),
-          p -> p.poller(Pollers.fixedDelay(resolvePollingTimeFrame(configuration.getAccessionDelay(), configuration.getAccessionTimeUnit()))))
+        .from(() -> {
+            var connectionSystemParameters = sms.getStagingDirectorConnectionParameters(configuration.getTenantId());
+            return remoteStorageService
+              .getAccessionQueueRecords(configuration.getId(), connectionSystemParameters.getTenantId(), connectionSystemParameters.getOkapiToken());
+          },
+          p -> p.poller(Pollers.fixedDelay(
```
Comment (VitaliiZahoruiko) [inline src/main/java/org/folio/ed/service/StagingDirectorFlowsService.java:92]: (minor)Please use more meaningful names for variables

**Review by VitaliiZahoruiko — APPROVED** [review]

**Review by siarhei-charniak — APPROVED** [review]

**Pull request merged**